docs/knowledge-base.rst

```diff
@@ -0,0 +1,107 @@
+.. _knowleedge-base:
+
+==============
+Knowledge base
+==============
+
+Genialis Knowledge base (KB) is a collection of "features" (genes,
+transcripts, exons, ...) and "mappings" between these features. It comes
+very handy when performing various tasks with genomic features e.g.:
+
+    - find all aliases of gene ``BRCA2``
+    - finding all genes of type ``protein_coding``
+    - find all transcripts of gene ``FHIT``
+    - converting ``gene_id`` to ``gene_symbol``
+    - ...
+
+
+Feature
+=======
+``Feature`` object represents a genomic feature: a gene, a transcript, an
+exon, etc. You can search and filter ``Feature`` objects by ``feature``
+endpoint, similarly like ``Data``, ``Sample`` or any other ReSDK resource::
+
+    feature1 = res.feature.get(feature_id="BRCA2")
+
+To examine all attributes of a ``Feature``, see the :ref:`reference`.
+Here we will list a few::
+
+    # Get the feature:
+    feature1 = res.feature.get(feature_id="BRCA2")
+
+    # Database where this feature is defined, e.g. ENSEMBL, UCSC, NCBI, ...
+    feature1.source
+
+    # Unique identifier of a feature
+    feature1.feature_id
+
+    # Feature species
+    feature1.species
+
+    # Feature type, e.g. gene, transcript, exon, ...
+    feature1.type
+
+    # Feature name
+    feature1.name
+
+    # List of feature aliases
+    feature1.aliases
+
+But the real power is in the search / filter capabilities. Let's just
+list some examples::
+
+    # Count number of Human "protein-conding" transcripts in ENSEMBL database
+    res.feature.filter(
+        species="Homo sapiens",
+        type="transcript",
+        subtype="protein-coding",
+        source="ENSEMBL",
+    ).count()
+
+    # Convert all gene_id's in a list ``gene_ids`` to gene symbols::
+    genes = res.feature.filter(
+        feature_id__in=gene_ids,
+        type="gene",
+        species="Homo sapiens",
+    )
+    gene_symbols = [g.name for g in genes]
```

Comment:
This is a good example of how not to do a conversion! I would actually keep this example in the docs with a big warning, so that the readers avoid this mistake. Because it's certainly tempting to do it like this. The problem, of course, is that there is no guarantee that `gene_symbols` will appear in the same order as `gene_ids`. I presume that the order is going to be the same as in the database, but @dblenkus probably knows much more about this. So I welcome that we show this example but of course also show how to do the conversion correctly.

Reply:
Good point, thanks. I've fixed the example.